File: src/main/java/com/sudosoo/takeiteasy/controller/HeartController.java
package com.sudosoo.takeiteasy.controller;

import com.sudosoo.takeiteasy.dto.heart.CommentHeartRequestDto;
import com.sudosoo.takeiteasy.dto.heart.PostHeartRequestDto;
import com.sudosoo.takeiteasy.service.HeartService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/heart")
@RequiredArgsConstructor
public class HeartController {
    private final HeartService heartService;

    @PostMapping(value = "/createPostHeart" , name = "createPostHeart")
    public ResponseEntity<Void> createPostHeart(@Valid @RequestBody PostHeartRequestDto requestDto) {

        heartService.createPostHeart(requestDto);

        return ResponseEntity.ok().build();
    }

    @PostMapping(value = "/createCommentHeart" , name = "createCommentHeart")
    public ResponseEntity<Void> createCommentHeart(@Valid @RequestBody CommentHeartRequestDto requestDto) {

        heartService.createCommentHeart(requestDto);

        return ResponseEntity.ok().build();
    }

    @PutMapping(value = "/createPostDisHeart" , name = "createPostDisHeart")
    public ResponseEntity<Void> createPostDisHeart(@Valid @RequestBody PostHeartRequestDto requestDto) {

        heartService.postDisHeart(requestDto);

        return ResponseEntity.ok().build();
    }

    @PutMapping(value = "/createCommentDisHeart" , name = "createCommentDisHeart")
    public ResponseEntity<Void> createCommentDisHeart(@Valid @RequestBody CommentHeartRequestDto requestDto) {

        heartService.commentDisHeart(requestDto);

        return ResponseEntity.ok().build();
    }
}

File: takeItEasy-presentation/src/main/kotlin/com/sudosoo/takeItEasy/presentation/controller/HeartController.kt
package com.sudosoo.takeItEasy.presentation.controller

import com.sudosoo.takeItEasy.application.dto.heart.CommentHeartRequestDto
import com.sudosoo.takeItEasy.application.dto.heart.PostHeartRequestDto
import com.sudosoo.takeItEasy.application.service.HeartService
import jakarta.validation.Valid
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/heart")
class HeartController (val heartService: HeartService){

    @PostMapping("/createPostHeart", name = "createPostHeart")
    fun createPostHeart(@RequestBody @Valid requestDto: PostHeartRequestDto): ResponseEntity<Void> {
        heartService.createPostHeart(requestDto)

        return ResponseEntity.ok().build<Void>()
    }

    @PostMapping("/createCommentHeart", name = "createCommentHeart")
    fun createCommentHeart(@RequestBody @Valid requestDto: CommentHeartRequestDto): ResponseEntity<Void> {
        heartService.createCommentHeart(requestDto)

        return ResponseEntity.ok().build<Void>()
    }

    @PutMapping("/createPostDisHeart", name = "createPostDisHeart")
    fun createPostDisHeart(@RequestBody @Valid requestDto: PostHeartRequestDto): ResponseEntity<Void> {
        heartService.postDisHeart(requestDto)

        return ResponseEntity.ok().build<Void>()
    }

    @PutMapping("/createCommentDisHeart", name = "createCommentDisHeart")
    fun createCommentDisHeart(@RequestBody @Valid requestDto: CommentHeartRequestDto): ResponseEntity<Void> {
        heartService.commentDisHeart(requestDto)

        return ResponseEntity.ok().build<Void>()
    }
}